lib/jsonld-signatures.js

```diff
@@ -283,35 +284,40 @@ api.verify = util.callbackify(async function(input, options) {
   let proofs = jsonld.getValues(framed, 'signature')
     .map(doc => ({property: 'signature', doc}))
     .concat(jsonld.getValues(framed, 'proof')
-      .map(doc => ({property: 'proof', doc})));
+    .map(doc => ({property: 'proof', doc})));
   if(proofs.length === 0) {
     throw new Error('No signature found.');
   }
 
   // Filter proofs to only those with the expected proofpurpose
-  const {purpose} = options;
-  if(purpose) {
-    const {proofPurpose} = purpose;
-    if(!proofPurpose) {
-      throw new Error('`purpose.proofPurpose` is required.');
-    }
-    const ProofPurpose = api.proofPurposes.use(proofPurpose);
-    if(!ProofPurpose) {
-      throw new Error(`Unsupported proof purpose handler "${proofPurpose}".`);
-    }
-    purpose.proofPurposeHandler = new ProofPurpose(injector);
+  const {purpose, purposeParameters} = options;
+  const SUPPORTED_PROOF_PURPOSES = _getSupportedProofPurposes();
+  if(SUPPORTED_PROOF_PURPOSES.indexOf(purpose) === -1) {
+    // TODO: Should we throw an error here? This will break all usages of
+    // json-ld signatures. Is it true that a proofPurpose is required for
+    // all signatures?
```

Comment:
Yes, we should require it. All signature proofs should have a purpose associated with them.

Reply:
If we need one iteration of a backwards compatible version, we can insert a default proof purpose that just checks against `publicKey` (not any special verification suite). But a major point release should remove this default behavior because it's dangerous.